DOC: update contributing to pandas documentation

marenwestermann opened the pull request:
- [x] closes #48035
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

I removed the part in the [Committing your code](https://pandas.pydata.org/docs/development/contributing.html#id9) section where it is said that commit messages should include the number of the GitHub issue the commit is contributing towards. I updated the pull request template to include cross references so that it's more clear how people should make use of referencing GitHub issues.

I fixed the code snippet `git commit -m` which causes an error and updated the documentation to include both commit message versions, the one line commit message and the more detailed commit message that can be done with an editor. Let me know if you would like to keep both versions or would like further changes.

Additionally, I moved the section about the prefixes to the [Finally, make the pull request](https://pandas.pydata.org/docs/development/contributing.html#id12) section of the documentation. Let me know if the prefixes need to be updated.

marenwestermann committed:
update contributing to pandas documentation

mroeschke commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,4 +1,5 @@
 - [ ] closes #xxxx (Replace xxxx with the Github issue number)
+- [ ] xref #xxxx (Replace xxxx with the Github issue number)
```
Why was this added? I am concern that it may be confusing with there also being `closes #xxxx above`

IMO we want to encourage the users to use `closes #xxxx` if applicable so it automatically closes the linked issue

phofl replied:
Yeah agreed. There are already many checkboxes you have to tick, would prefer to not add this

datapythonista commented on doc/source/development/contributing.rst:
```diff
@@ -194,31 +194,18 @@ Doing 'git status' again should give something like::
-relevant GitHub issues in your commit message using GH1234 or #1234.  Either style
-is fine, but the former is generally preferred:
+Finally, commit your changes to your local repository with an explanatory commit
+message of ``< 80`` chars::
```
```suggestion
message::
```
I know this already exists now, but since we squash commits, not very important, and we can simplify things a bit.

datapythonista commented on doc/source/development/contributing.rst:
```diff
@@ -194,31 +194,18 @@ Doing 'git status' again should give something like::
+
+    git commit -m "your commit message goes here"
+
+Alternatively, you can just type ``git commit`` which opens an editor and use the
+following commit message structure:
 
 * a subject line with ``< 80`` chars.
 * One blank line.
 * Optionally, a commit message body.
```
I'd just delete all this.

datapythonista reviewed (commented):
Thanks @marenwestermann, nice update. Added couple of comments of things that I'd change. In the current docs, your changes look fine other than the template changes already mentioned.

marenwestermann committed:
address requested changes

marenwestermann commented:
Thanks for your feedback! I addressed your comments. I added the line `- [ ] xref #xxxx (Replace xxxx with the Github issue number)` in `PULL_REQUEST_TEMPLATE.md` because sometimes a pull request contributes to an issue but doesn't close it such as here: https://github.com/pandas-dev/pandas/pull/44798. However, I understand your concerns and removed this line.

marenwestermann committed:
fix typo

datapythonista reviewed (approved):
lgtm, thanks @marenwestermann, nice improvement.

mroeschke commented:
Thanks @marenwestermann!
